src/components/ConnectionHelp.js

```diff
@@ -205,26 +256,57 @@ class ConnectionHelp extends Component {
             <PaddedBox display="flex" justifyContent="center">
               <RoverConnection />
             </PaddedBox>
-            <Typography gutterBottom>
-              <FormattedMessage
-                id="app.connection_help.done"
-                description="tells the user how to select their micro:bit BLE device"
-                defaultMessage="Then, choose the BBC micro:bit device from the list and click Pair.
-                If the Rover icon turns blue, you're connected! Click outside this
-                window, and try the first lesson!"
-              />
-            </Typography>
+            {
+              rover ? (
+                <PaddedBox display="flex" justifyContent="center">
+                  <Button
+                    color="primary"
+                    variant="outlined"
+                    onClick={this.handleFinishedClick}
+                  >
+                    <FormattedMessage
+                      id="app.connection_help.to_courses"
+                      description="Links to courses"
+                      defaultMessage="Congrats, you're connected! Click here, and let's write some code."
+                    />
+                  </Button>
+                </PaddedBox>
+              ) : (
+                <Typography gutterBottom>
+                  <FormattedMessage
+                    id="app.connection_help.done"
+                    description="tells the user how to select their micro:bit BLE device"
+                    defaultMessage="Then, choose the BBC micro:bit device from the list and click Pair."
+                  />
+                </Typography>
+              )
+            }
+            {
+              done ? (
+                <Redirect to={{
+                  pathname: '/programs/mine',
+                }}
+                />
+              ) : (null)
+            }
           </DialogContent>
         </Dialog>
       </>
     );
   }
 }
 
+ConnectionHelp.defaultProps = {
+  rover: null,
+};
+
 ConnectionHelp.propTypes = {
   user: PropTypes.shape({
     // showConnectionHelpOnLogin: PropTypes.bool.isRequired, // TODO
   }).isRequired,
+  rover: PropTypes.shape({
```

Comment:
I don't think there should be a case where `rover` isn't present since it's coming from redux. Could this be made `isRequired`?

Reply:
This is the `rover.rover` from `mapStateToProps = ({ user, rover }) => ({ user, rover: rover.rover })`, right? That one could be null.

Reply:
Oh, but `null` is still present? I'll make it required.

Reply:
Ah, I think it can't be required: 
```
Failed prop type: The prop `rover` is marked as required in `ConnectionHelp`, but its value is `null`.
```

Reply:
Ok. That answers it then